include/nbl/builtin/hlsl/workgroup/shared_scan.hlsl

```diff
@@ -75,72 +73,59 @@ struct WorkgroupScanHead
             {
                 const uint prevLevelScan = sharedAccessor.main.get(gl_LocalInvocationIndex);
                 scan = subgroupOp(prevLevelScan);
-                if(isScan)
-                {
-                    sharedAccessor.main.set(scanStoreIndex, scan);
-                }
-            }
-            if(isScan)
-            {
-                scanStoreIndex += lastInvocationInLevel + 1u;
+                sharedAccessor.main.set(scanLoadIndex, scan);
```

Comment:
THIS IS STILL WRONG

Reply:
the `set` is not supposed to be here!
you're supposed to `get(scanLoadIndex)` !

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/519#discussion_r1327248553

Reply:
here's a suggestion, replace everything after and including `bool participate` with
```cpp
        bool participate = gl_LocalInvocationIndex <= lastInvocationInLevel;
        scanLoadIndex -= lastInvocationInLevel-1u; // to cancel out the index shift on the first iteration
        // TODO: later [unroll(scan_levels<WorkgroupSize,MinSubgroupSize>::value)]
        [unroll(1)]
        while(lastInvocationInLevel >= glsl::gl_SubgroupSize())
        {
            scanLoadIndex += lastInvocationInLevel+1u;
            // only invocations that have the final value of the subgroupOp (inclusive scan) store their results
            if (participate && (gl_LocalInvocationIndex==lastInvocationInLevel || isLastSubgroupInvocation))
                sharedAccessor.main.set(scanLoadIndex - loadStoreIndexDiff, scan); // For subgroupSz = 32, first 512 invocations store index is [0,15], 512-1023 [16,31] etc.
            sharedAccessor.main.workgroupExecutionAndMemoryBarrier();
            participate = gl_LocalInvocationIndex <= (lastInvocationInLevel >>= glsl::gl_SubgroupSizeLog2());
            if(participate)
            {
                const uint prevLevelScan = sharedAccessor.main.get(scanLoadIndex);
                scan = subgroupOp(prevLevelScan);
            }
        }
        lastLevelScan = scan; // only invocations of gl_LocalInvocationIndex < gl_SubgroupSize will have correct values, rest will have garbage
```